pkg/services/ngalert/store/alert_rule_state.proto

```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+package state;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "./pb";
+
+message AlertInstanceKey {
+    int64 rule_org_id = 1;
+    string rule_uid = 2;
+    string labels_hash = 3;
+}
+
+message AlertInstance {
+    AlertInstanceKey key = 1;
+    map<string, string> labels = 2;
+    string current_state = 3;
+    string current_reason = 4;
+    google.protobuf.Timestamp current_state_since = 5;
+    google.protobuf.Timestamp current_state_end = 6;
+    google.protobuf.Timestamp last_eval_time = 7;
+    google.protobuf.Timestamp last_sent_at = 8;
+    google.protobuf.Timestamp resolved_at = 9;
+    string result_fingerprint = 10;
+}
+
+message AlertInstances {
+    repeated AlertInstance instances = 1;
```

Comment:
I wonder if it would be simpler to just keep it a map?

Reply:
Ah, it will change the interface of AlertInstanceStore, which we do not want to do. Nevermind